runtime/near-vm-logic/src/logic.rs

```diff
@@ -943,14 +945,131 @@ impl<'a> VMLogic<'a> {
+            .ok_or(VMLogicError::HostError(HostError::IntegerOverflow))?
+            / 64
+            + 1;
+
+        self.gas_counter.pay_per(ripemd160_block, message_blocks as u64)?;
+
+        use ripemd160::Digest;
+
+        let value_hash = ripemd160::Ripemd160::digest(&value);
+        self.internal_write_register(register_id, value_hash.as_slice().to_vec())
+    }
+
+    /// Recovers an ECDSA signer address and returns it into `register_id`.
+    ///
+    /// Takes in an additional flag to check for malleability of the signature
+    /// which is generally only ideal for transactions.
+    ///
+    /// Returns a bool indicating success or failure as a `u64`.
+    ///
+    /// # Malleability Flags
+    ///
+    /// 0 - No extra checks.
+    /// 1 - Rejecting upper range.
+    ///
+    /// # Errors
+    ///
+    /// * If `hash_ptr`, `r_ptr`, or `s_ptr` point outside the memory or the registers use more
+    ///   memory than the limit, then returns `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * 64 + ecrecover_base`
+    #[cfg(feature = "protocol_feature_math_extension")]
+    pub fn ecrecover(
+        &mut self,
+        hash_len: u64,
+        hash_ptr: u64,
+        sig_len: u64,
+        sig_ptr: u64,
+        v: u64,
+        malleability_flag: u64,
+        register_id: u64,
+    ) -> Result<u64> {
+        self.gas_counter.pay_base(ecrecover_base)?;
+
+        let signature = {
+            let vec = self.get_vec_from_memory_or_register(sig_ptr, sig_len)?;
+            if vec.len() != 64 {
+                return Err(VMLogicError::HostError(HostError::ECRecoverError {
+                    msg: format!(
+                        "The length of the signature: {}, exceeds the limit of 64 bytes",
+                        vec.len()
+                    ),
+                }));
+            }
+
+            let mut bytes = [0u8; 65];
+            bytes[0..64].copy_from_slice(&vec);
+
+            if v < 4 {
+                bytes[64] = v as u8;
+                Secp256K1Signature::from(bytes)
+            } else {
+                return Err(VMLogicError::HostError(HostError::ECRecoverError {
+                    msg: format!("V recovery byte 0 through 3 are valid but was provided {}", v),
+                }));
+            }
+        };
+
+        let hash = {
+            let vec = self.get_vec_from_memory_or_register(hash_ptr, hash_len)?;
+            if vec.len() != 32 {
+                return Err(VMLogicError::HostError(HostError::ECRecoverError {
+                    msg: format!(
+                        "The length of the hash: {}, exceeds the limit of 32 bytes",
+                        vec.len()
+                    ),
+                }));
+            }
+
+            let mut bytes = [0u8; 32];
+            bytes.copy_from_slice(&vec);
+            bytes
+        };
+
+        if !signature.check_signature_values((malleability_flag & 1) == 1) {
```

Comment:
Please check that `malleability_flag` is 0 or 1.

Reply:
Fixed in https://github.com/near/nearcore/pull/4380/commits/73380e1f1d565bb8c97e1b1b87b2710e71560221